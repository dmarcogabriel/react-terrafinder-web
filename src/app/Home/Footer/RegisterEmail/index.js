import React, { useState } from 'react';
import { container } from './RegisterEmail.module.scss';

export default function RegisterEmail() {
  const [email, setEmail] = useState('');

  function registerEmail() {
    // todo: validate email
    // todo: send email to api
  }

  return (
    <form className={container}>
      <input
        type="email"
        testID="emailInput"
        placeholder="EndereÃ§o de email"
        value={email}
        onChange={(e) => setEmail(e.target.value)}
      />

      <button type="button" testID="registerEmail" onClick={registerEmail}>
        Cadastrar
      </button>
    </form>
  );
}
