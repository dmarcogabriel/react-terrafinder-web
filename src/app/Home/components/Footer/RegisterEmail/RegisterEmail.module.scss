@import 'styles';

$padding: 0.6rem;
$font-size: 0.8rem;

.container {
  margin: 2rem 0;
  display: inline-block;
  border: 1px solid rgba(255, 255, 255, 0.5);
  border-radius: 0.25rem;
  overflow: hidden;

  > input {
    padding: $padding;
    background: transparent;
    font-size: $font-size;
    color: $white;

    &::placeholder {
      color: $white;
      opacity: 0.5;
    }
  }

  > button {
    padding: $padding;
    font-size: $font-size;
    background: $green;
    color: $white;
  }
}
