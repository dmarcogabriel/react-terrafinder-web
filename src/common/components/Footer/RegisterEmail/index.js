import React, { useState } from 'react';
import api from 'services/api';
import { useNotification, NOTIFICATION_TYPES } from 'hooks/useNotification';
import { container } from './RegisterEmail.module.scss';

export default function RegisterEmail() {
  const [email, setEmail] = useState('');
  const { showNotification } = useNotification();

  const registerEmail = async () => {
    try {
      await api.post('password/forgot', { email });
      showNotification(
        'Email de recuperação de senha enviado.',
        NOTIFICATION_TYPES.SUCCESS
      );
    } catch (error) {
      showNotification('Erro ao enviar email.', NOTIFICATION_TYPES.ERROR);
    }
  };

  return (
    <form className={container}>
      <input
        type="email"
        data-testid="emailInput"
        placeholder="Endereço de email"
        value={email}
        onChange={(e) => setEmail(e.target.value)}
      />

      <button type="button" data-testid="registerEmail" onClick={registerEmail}>
        Cadastrar
      </button>
    </form>
  );
}
