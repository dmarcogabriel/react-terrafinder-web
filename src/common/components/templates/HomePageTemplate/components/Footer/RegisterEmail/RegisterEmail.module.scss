@import 'styles';

$padding: 0.6rem;
$font-size: 0.8rem;

.container {
  margin: 2rem 0;
  border: 1px solid rgba(255, 255, 255, 0.5);
  border-radius: 0.25rem;
  overflow: hidden;
  display: inline-block;

  > input {
    padding: $padding;
    background: transparent;
    font-size: $font-size;
    color: $white;

    &::placeholder {
      color: $white;
      opacity: 0.5;
    }
  }

  > button {
    padding: $padding;
    font-size: $font-size;
    background: $green;
    color: $white;
    cursor: pointer;
  }
}
