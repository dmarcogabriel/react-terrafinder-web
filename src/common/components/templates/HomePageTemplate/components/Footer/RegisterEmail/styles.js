import { styled } from '@mui/material/styles';
import { InputBase, Typography } from '@mui/material';

export const RegisterEmailTitle = styled(Typography)({
  fontWeight: 'bold',
});

export const EmailInput = styled(InputBase)(({ error }) => {
  if (error)
    return {
      color: 'red',
    };

  return {};
});
