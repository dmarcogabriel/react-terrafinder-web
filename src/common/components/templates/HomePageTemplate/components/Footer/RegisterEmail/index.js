import React from 'react';
import { Paper, Button, Typography, Box } from '@mui/material';
import { useFormik } from 'formik';
import { object, string } from 'yup';
import { useNotification, NOTIFICATION_TYPES } from 'hooks/useNotification';
import { EmailInput, RegisterEmailTitle } from './styles';

export const RegisterEmail = () => {
  const { showNotification } = useNotification();
  const { handleChange, handleSubmit, values, errors, isValid } = useFormik({
    initialValues: {
      email: '',
    },
    validationSchema: object().shape({
      email: string().email().required(),
    }),
    async onSubmit({ email }) {
      try {
        console.log('%c[Email]: ', 'color: cyan', email);
        // todo: send email to api
        showNotification(
          'E-mail enviado com sucesso!',
          NOTIFICATION_TYPES.SUCCESS
        );
      } catch (error) {
        showNotification(
          'Desculpe, um erro ocorreu, tente novamente mais tarde.',
          NOTIFICATION_TYPES.ERROR
        );
      }
    },
  });

  return (
    <Box sx={{ maxWidth: { md: '30%' } }}>
      <RegisterEmailTitle variant="h6">
        Fique por dentro das novidades!
      </RegisterEmailTitle>
      <Typography sx={{ my: 1 }}>
        Se inscreva em nossa newsletter para receber os anúncios com os melhores
        preços!
      </Typography>
      <Paper
        component="form"
        sx={{
          display: 'flex',
          alignItems: 'center',
          mt: 2,
          overflow: 'hidden',
        }}
        noValidate
        onSubmit={handleSubmit}
      >
        <EmailInput
          error={!!errors.email}
          value={values.email}
          fullWidth
          sx={{ ml: 1 }}
          inputProps={{ inputMode: 'email', 'data-testid': 'emailInput' }}
          onChange={({ target: { value } }) => handleChange('email')(value)}
          placeholder="Informe seu e-mail aqui"
        />
        <Button
          data-testid="registerEmail"
          disabled={!isValid}
          variant="contained"
          onClick={handleSubmit}
          color="success"
          sx={{ borderRadius: 0 }}
        >
          Cadastrar
        </Button>
      </Paper>
    </Box>
  );
};
