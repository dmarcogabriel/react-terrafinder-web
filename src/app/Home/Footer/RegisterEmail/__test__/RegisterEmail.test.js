import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import RegisterEmail from '..';

describe('<RegisterEmail />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RegisterEmail />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Add tests here...

  it('matches snapshot', () => {
    const tree = renderer.create(<RegisterEmail />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
