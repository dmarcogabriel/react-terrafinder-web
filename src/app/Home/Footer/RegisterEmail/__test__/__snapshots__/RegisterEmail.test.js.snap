// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterEmail /> matches snapshot 1`] = `
<form>
  <input
    onChange={[Function]}
    placeholder="EndereÃ§o de email"
    type="email"
    value=""
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Cadastrar
  </button>
</form>
`;
